google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/admin/v2/it/BigtableCmekIT.java

```diff
@@ -0,0 +1,243 @@
+import static org.junit.Assert.fail;
+
+import com.google.api.gax.rpc.ApiException;
+import com.google.api.gax.rpc.NotFoundException;
+import com.google.cloud.bigtable.admin.v2.BigtableInstanceAdminClient;
+import com.google.cloud.bigtable.admin.v2.BigtableTableAdminClient;
+import com.google.cloud.bigtable.admin.v2.internal.NameUtil;
+import com.google.cloud.bigtable.admin.v2.models.Backup;
+import com.google.cloud.bigtable.admin.v2.models.Cluster;
+import com.google.cloud.bigtable.admin.v2.models.CreateBackupRequest;
+import com.google.cloud.bigtable.admin.v2.models.CreateClusterRequest;
+import com.google.cloud.bigtable.admin.v2.models.CreateInstanceRequest;
+import com.google.cloud.bigtable.admin.v2.models.CreateTableRequest;
+import com.google.cloud.bigtable.admin.v2.models.EncryptionInfo;
+import com.google.cloud.bigtable.admin.v2.models.StorageType;
+import com.google.cloud.bigtable.common.Status;
+import com.google.cloud.bigtable.common.Status.Code;
+import com.google.cloud.bigtable.test_helpers.env.AbstractTestEnv;
+import com.google.cloud.bigtable.test_helpers.env.EmulatorEnv;
+import com.google.cloud.bigtable.test_helpers.env.TestEnvRule;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.function.ThrowingRunnable;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.threeten.bp.Instant;
+import org.threeten.bp.temporal.ChronoUnit;
+
+/**
+ * Tests our CMEK offering. It can take up to 5 mins after a CMEK-protected table is created for the
+ * key version and status fields to be populated. Set the {@code bigtable.wait-for-cmek-key-status}
+ * system property while running the test in order to test the populated key fields by incurring a
+ * wait.
+ */
+@RunWith(JUnit4.class)
+public class BigtableCmekIT {
+
+  private static final String CMEK_KMS_KEY_PROPERTY_NAME = "bigtable.kms_key_name";
+  private static final int[] BACKOFF_DURATION = {5, 10, 50, 100, 150, 200, 250, 300};
+  private static final Logger LOGGER = Logger.getLogger(BigtableCmekIT.class.getName());
+  private static final String TEST_TABLE_ID = "test-table-for-cmek-it";
+
+  @ClassRule
+  public static TestEnvRule testEnvRule = new TestEnvRule();
+
+  private static String kmsKeyName;
+  private String instanceId;
+  private String clusterId;
+
+  private BigtableInstanceAdminClient instanceAdmin;
+  private BigtableTableAdminClient tableAdmin;
+
+  @BeforeClass
+  public static void validatePlatform() {
+    assume()
+        .withMessage("Emulator doesn't support on CMEK")
+        .that(testEnvRule.env())
+        .isNotInstanceOf(EmulatorEnv.class);
+
+    kmsKeyName = System.getProperty(CMEK_KMS_KEY_PROPERTY_NAME);
+
+    assume().withMessage("%s property is not set, skipping CMEK test", kmsKeyName);
```

Comment:
please refactor this to never skip as it should now run with every other IT

Reply:
Is it okay to load it as an optional property like we do now? Would be cumbersome to alway specify this when running other tests individually.

Reply:
commented on other thread

Reply:
Added the env variable to integration.cfg files.

Reply:
We should still remove this assertion. It makes it very hard to notice if tests are actually running or getting skipped in the logs. Alternatively, you can add a new assertion that asserts the `kmsKeyName` is not null, and fails the test if so.

Reply:
Thanks for taking care of this!